scripts/calendar_import.py

```diff
@@ -4,20 +4,26 @@
 # For modifications please refer to the Google Calendar Python API:
 # https://developers.google.com/resources/api-libraries/documentation/calendar/v3/python/latest/calendar_v3.events.html#insert
 
+# Requires the following packages: oauth2client, pyyaml, google-api-python-client
+
+# Uses a service account -- the calendar must be shared with the service account
+# email and given permission: "Make changes to events"
+
 from datetime import datetime
 import logging
 import os.path
 import yaml
 import googleapiclient.errors
 from googleapiclient.discovery import build
-from google_auth_oauthlib.flow import InstalledAppFlow
+from oauth2client.service_account import ServiceAccountCredentials
 
+SERVICE_ACCOUNT_FILE = 'kubeflow_calendar_sa.json'
```

Comment:
Why are you hard coding the service_account file.

The convention is to look for a GOOGLE_APPLICATION_CREDENTIALS environment variable which is pointing at the key.

Moving forward though we'd actually like to use GKE Workload identity;
https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity

Do you know if this code will work with Workload Identity?

Reply:
I read over that page and how I understand it is that a service account is available to a pod... so I'm assuming the service account could still be referred to as `GOOGLE_APPLICATION_CREDENTIALS`? If so, that should work.

Reply:
When workload identity is used the identity will not be determined by GOOGLE_APPLICATION_CREDENTIALS

Instead the pod should have a KSA (Kubernetes Service Account) that is mapped to a GSA(Google Service Account).  Google client libraries should be initiated to use the application default credentials in which case the credential should map to whatever GSA is mapped to the KSA.

At any rate I think we can figure it out when we get there.